badge_server/main.py

```diff
@@ -75,519 +88,235 @@ def _get_pair_status_for_packages(pkg_sets):
-
-                        # Ignore the package that are not self compatible
-                        self_status = _get_result_from_cache(
-                            package_name=package_name,
-                            badge_type=badge_utils.BadgeType.SELF_COMP_BADGE)
-                        if self_status[py_version]['status'] not in [
-                                'SUCCESS', 'CALCULATING']:
-                            continue
-                        # Status showing one of the check failures
-                        version_and_res[
-                            py_version]['status'] = res_item.get('result')
-                        description = res_item.get('description')
-                        details = badge_utils.EMPTY_DETAILS if description \
-                            is None else description
-                        version_and_res[
-                            py_version]['details'][package] = details
-            version_and_res['timestamp'] = datetime.datetime.now().strftime(
-                badge_utils.TIMESTAMP_FORMAT)
-            result = version_and_res
-
-        # Write the result to Cloud Datastore
-        cache.set('{}_google_comp_badge'.format(package_key), result)
-
-    google_comp_res = cache.get('{}_google_comp_badge'.format(package_key))
-
-    if not utils._is_package_in_whitelist([package_name]):
-        google_comp_res = badge_utils._build_default_result(
-            badge_type=badge_utils.BadgeType.GOOGLE_COMP_BADGE,
-            status='UNKNOWN',
-            details={})
-
-    if google_comp_res is None:
-        details = badge_utils._build_default_result(
-            badge_type=badge_utils.BadgeType.GOOGLE_COMP_BADGE,
-            status='CALCULATING',
-            details={})
-    else:
-        details = google_comp_res
-
-    # Run the check if google_comp_res is None or forced to populate the cache
-    # or the cache is outdated.
-    if google_comp_res is None or force_run_check is not None:
-        threading.Thread(target=run_check).start()
-    elif google_comp_res is not None:
-        timestamp = google_comp_res.get('timestamp')
-        if not badge_utils._is_github_cache_valid(timestamp):
-            threading.Thread(target=run_check).start()
-
-    badge = badge_utils._get_badge(details, badge_name)
-    response = flask.make_response(badge)
-    response.content_type = badge_utils.SVG_CONTENT_TYPE
+    # Set the cache for force refreshing the badge image when refreshing the
+    # github readme page, otherwise the image will stay the same and won't be
+    # updated for a while.
     response.headers['Cache-Control'] = 'no-cache'
+
+    # https://tools.ietf.org/html/rfc2616#section-13.4 allows success responses
+    # to be cached if no `Cache-Control` header is set. Since the content of
+    # the image is frequently updated, caching is explicitly disabled to force
+    # the client/cache to refetch the content on every request.
     response.add_etag()
```

Comment:
I think that comment applies to the previous line and I still don't know what the etag does.

Reply:
I was going off an [earlier comment](https://github.com/GoogleCloudPlatform/cloud-opensource-python/pull/257#discussion_r261708169) you made, which I had thought was directed towards the `etag` line, but now I'm confused

Reply:
Summary of 1:1 conversation:
`add_etag()` line should have been removed.

Reply:
done